Content.Server/Materials/MaterialReclaimerSystem.cs

```diff
@@ -258,9 +258,24 @@ private void SpawnChemicalsFromComposition(EntityUid reclaimer,
         }
 
         // if the item we inserted has reagents, add it in.
-        if (_solutionContainer.TryGetDrainableSolution(item, out _, out var drainableSolution))
+
+        if (reclaimerComponent.OnlyReclaimDrainable)
         {
-            totalChemicals.AddSolution(drainableSolution, _prototype);
+            // Are we a recycler? Only use drainable solutions.
+            if (_solutionContainer.TryGetDrainableSolution(item, out _, out var drainableSolution))
+            {
+                totalChemicals.AddSolution(drainableSolution, _prototype);
+            }
+        }
+        else
+        {
+            // Are we an industrial reagent grinder? Use any solutions it has.
+            if (TryComp<SolutionContainerManagerComponent>(item, out var solutionContainer)){
+                foreach (var (_, soln) in _solutionContainer.EnumerateSolutions((item, solutionContainer)))
+                {
+                    totalChemicals.AddSolution(soln.Comp.Solution, _prototype);
+                }
+            }
```

Comment:
this really should be getting the grindable or juiceable solutions rather than directly enumerating all of them.

Reply:
Done. Grindable was specifically chosen because it is the industrial reagent _grinder_, and I imagine the major use case of this will be in chem where most things, even if they can be juiced, are preferable to grind, especially since cross-breeding affects the grindable solution and not the juiceable one. Exceptions can be juiced in the old grinder, or someone can add an industrial reagent juicer or a mode toggle if they really want to.